weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallPage/OverflowMenu.tsx

```diff
@@ -102,28 +112,36 @@ const ConfirmDeleteModal: FC<{
 
   const [deleteLoading, setDeleteLoading] = useState(false);
 
-  let error = null;
-  if (calls.length === 0) {
-    error = 'No calls selected';
-  } else if (new Set(calls.map(c => c.entity)).size > 1) {
-    error = 'Cannot delete calls from multiple entities';
-  } else if (new Set(calls.map(c => c.project)).size > 1) {
-    error = 'Cannot delete calls from multiple projects';
-  } else if (calls.length > 1) {
-    error = 'Cannot delete multiple calls';
-  }
+  const [error, setError] = useState<string | null>(null);
 
   const onDelete = () => {
+    if (calls.length === 0) {
+      setError('No call(s) selected');
+      return;
+    }
+    // error if all projects are not the same
+    if (calls.every(call => call.project !== calls[0].project)) {
+      setError('All calls must be in the same project to delete');
+      return;
+    }
     setDeleteLoading(true);
     callsDelete(
       calls[0].entity,
       calls[0].project,
-      calls.map(c => c.callId)
-    ).then(() => {
-      setDeleteLoading(false);
-      setConfirmDelete(false);
-      closePeek();
-    });
+      calls.map(call => call.callId)
+    )
+      .catch(() => {
+        setError(
+          `Failed to delete call(s) ${calls
+            .map(call => call.displayName ?? call.spanName)
+            .join(', ')}`
+        );
```

Comment:
not sure what happens if deletion fails in the middle... might just want to error ambigously